src/split.c

```diff
@@ -111,11 +112,8 @@ rcutils_split(
   return RCUTILS_RET_OK;
 
 fail:
-  error_msg = "unable to allocate memory for string array data";
-  if (rcutils_string_array_fini(string_array) != RCUTILS_RET_OK) {
-    error_msg = rcutils_format_string(allocator, "FATAL: %s. Leaking memory", error_msg);
-  }
-  RCUTILS_SET_ERROR_MSG(error_msg, allocator);
+  RCUTILS_SET_ERROR_MSG("unable to allocate memory for string array data", allocator);
+  rcutils_string_array_fini(string_array);
```

Comment:
I don't think this is a good idea to not check the return value. If it is not `RCUTILS_RET_OK` then it will have set the error message internally, overwriting the error message set one line above.

I'd do something like this:

```c++
if (rcutils_string_array_fini(string_array) != RCUTILS_RET_OK) {
  RCUTILS_SAFE_FWRITE_TO_STDERR("failed to finalize string array during error handling: ");
  RCUTILS_SAFE_FWRITE_TO_STDERR(rcutils_get_error_string_safe());
  RCUTILS_SAFE_FWRITE_TO_STDERR("\n");
  rcutils_reset_error();
}
RCUTILS_SET_ERROR_MSG("unable to allocate memory for string array data", allocator);
```